lib/spack/llnl/util/symlink.py

```diff
@@ -24,7 +24,7 @@ def symlink(real_path, link_path):
     On Windows, use junctions if os.symlink fails.
     """
     if not is_windows or _win32_can_symlink():
-        os.symlink(real_path, link_path)
+        os.symlink(real_path, link_path, target_is_directory=os.path.isdir(real_path))
```

Comment:
Why was this added? It's a really bad idea, cause this way dangling symlinks cannot be created.

Reply:
> In general I'd like to ask you to never introduce changes for Windows support that cause performance regressions for non-Windows.

This request is impossible to satisfy, or even to break down into something realistic or manageable. It is equivalent to saying "don't introduce bugs" (I'm already trying to do that).

As for the specific issue, I personally think a mistake at this level is unavoidable. Even if your original complaint was true. We have this code, and then we have you noticing and making an adjustment that avoids the issue (presumably because you have a benchmark which detects that something started slowing down). If you're worried about dangling symlinks and the tests don't have it, then they should have it; if you think there should be a benchmark for tracking this type of slowdown and we don't have it, then we should have one.